src/psyclone/psyir/nodes/omp_directives.py

```diff
@@ -511,6 +585,14 @@ def gen_code(self, parent):
 
         self.gen_post_region_code(parent)
 
+    def lower_to_language_level(self):
+        ''' In-place construction of clauses as PSyIR constructs.'''
```

Comment:
Can you extend this description to explain that in the higher level some directives relay on dynamic variable dependence logic to decide what is private/shared and then we use this lowering step to find out this references and place them explicitly to the lower level tree the the backend visitor will process.

Reply:
Done.